### PR TITLE
Workshop

### PR DESCRIPTION
**Pull request by artur-rs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,39 @@
...
+`README`
+
+Presentation
+------------
+
+RTE-Workshop presentation is created with Remark - in-browser, markdown-driven slideshow tool. To run such presentation, clone this repository, then go to
+`rte-workshop` directory with all required files and run:
+
+```
+sudo python -m SimpleHTTPServer 80
```
Comment (pietrushnic) [inline README.md]: @artur-rs please add `python3 -m http.server 80`, also I'm not sure if default port 80 is ok, people may have different services at this port maybe `8000`, which should be default is good enough?

Comment (pietrushnic) [reply README.md]: @artur-rs python3 as alternative

File: README.md [hunk 2]
```diff
@@ -1,2 +1,39 @@
...
+```
+Now, open your favourite browser and type `0.0.0.0`. There should be listed
+directory tree with `rte-workshop.html` file. Just click it and enjoy our
+presentation.
```
Comment (pietrushnic) [inline README.md]: @artur-rs maybe screen here?

File: flash-apu.robot [hunk 3]
```diff
@@ -0,0 +1,27 @@
+*** Settings ***
```
Comment (pietrushnic) [inline flash-apu.robot]: @artur-rs don't you think all robot scripts should be in some directory?

**Comment by pietrushnic** [comment]
@artur-rs I think presentation and workshop scripts should be placed in distinguishing directories eg. `presentations`, `robot` or `rf-scripts`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,39 @@
...
+user:       rte-workshop
+password:   rte-workshop
+```
+* all necessary files can be found in `~/RTE-Workshop/` directory including
+`README`
+
+Presentation
+------------
+
+RTE-Workshop presentation is created with Remark - in-browser, markdown-driven slideshow tool. To run such presentation, clone this repository, then go to
```
Comment (miczyg1) [inline README.md]: line over 80 characters

File: rte-workshop.md [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+.center[<img src="images/rte-setup-diagram-tp.png" width="500px"/>]
+
+Additional infrastructure that can be used is Debian machine with iPXE server
+for supplementary boot option.
+
+---
+
+# Hardware setup
+
+* connect RTE Hat with Orange Pi Zero board (~1.5min)
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs we have more time (90min) - let's make it 3min

File: rte-workshop.md [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+
+.center[![HAT with OPi](images/conn-hat.jpg)]
+
+---
+
+# Hardware setup
+
+* RS232 interface via a null modem cable (~1min)
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs 2min

File: rte-workshop.md [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+
+.center[![RS232](images/conn-rs232.jpg)]
+
+---
+
+# Hardware setup
+
+* SPI interface via IDC 2x4 pin cable (~1.5min)
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs 3min

File: rte-workshop.md [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+
+.center[.inner[
+<img src="images/spi-apu-rte.png" height="400px">
+![IDC](images/conn-idc.jpg)]]
+
+---
+
+# Hardware setup
+
+* APU PWR/RST pins with RTE OC buffers (~3min)
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs 5min

File: rte-workshop.md [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+
+.left[![Other pins](images/conn-other2.png)]
+
+---
+
+# Hardware setup
+
+* DC-DC jack cable (~30s)
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs 2min

File: rte-workshop.md [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+
+.center[![DC Jack](images/conn-dcjack.jpg)]
+
+---
+
+# Hardware setup
+
+.center[<img src="images/full-setup.png" width="500px">]
+
+* take the connected setup to the power/network section (~6min):
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs 10min - please make all times bold

File: rte-workshop.md [hunk 11]
```diff
@@ -0,0 +1,356 @@
...
+
+---
+
+# Automation tests - RobotFramework
+
+* Generic test automation framework  for acceptance test-driven development
+(ATDD)
+* Utilizes the keyword-driven testing approach
+* The core is implemented using Python and runs also on Jython (JVM) and
+IronPython (.NET)
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs at the point about Test Driven Bug Fixing (TDBF)

File: rte-workshop.md [hunk 12]
```diff
@@ -0,0 +1,356 @@
...
+
+*** Test Cases ***
+RTE: 1.1 SSH connection
+    ${ssh_info}=    SSHLibrary.Get Connection
+    Should Be Equal As Strings    ${ssh_info.host}    ${rte_ip}
+```
+
+---
+
+# Automation tests - setup environment
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs this slide missing time, let's give some arbitrary time - 10min?

File: rte-workshop.md [hunk 13]
```diff
@@ -0,0 +1,356 @@
...
+
+* **`robot -v rte_ip:RTE_IP validation.robot`**
+
+.center[![validation cmd](images/cmd2.png)]
+
+where **`RTE_IP`** is your RTE IP address
+
+--
+
+.center[<img src="images/cmd3.png" width="700px"/>]
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs time?

File: rte-workshop.md [hunk 14]
```diff
@@ -0,0 +1,356 @@
...
+    * **NoValidConnections** - bad IP address
+    * **Telnet timeout** - check RS232 cable or close established connection
+
+--
+
+* After each RobotFramework test suite we can track generated log files:
+
+.left[![validation result](images/cmd5.png)]
+
+Try it by yourself by typing: **`firefox log.html`**
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs what about other log files?

File: rte-workshop.md [hunk 15]
```diff
@@ -0,0 +1,356 @@
...
+* remember to change **fw_file** firmware path for appropriate APU2/3/4 binary
+(if you are not sure, check the sign of life from previous tests)
+
+.center[![flash cmd](images/cmd6.png)]
+
+--
+
+Possible failures:
+* **opening "/tmp/coreboot.rom" failed** - SSH key not authenticated
+* **No such file or directory** - bad firmware file path
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs time?

File: rte-workshop.md [hunk 16]
```diff
@@ -0,0 +1,356 @@
...
+
+---
+
+# Fixing bugs
```
Comment (pietrushnic) [inline rte-workshop.md]: @artur-rs time?

**Review by pietrushnic — CHANGES_REQUESTED** [review]

**Comment by pietrushnic** [comment]
@artur-rs it looks like you pushed stuff, please check my comments which are outdated right now.

**Comment by pietrushnic** [comment]
@artur-rs all slides with tasks for attendees shall have timing.

File: presentations/rte-workshop.md [hunk 17]
```diff
@@ -0,0 +1,356 @@
...
+* Problem with automation and remote development
+
+<br/>
+To get rid of these problems we created Remote Testing Environment.
+
+---
+
+# Remote Testing Environment
+
+.center[<img src="images/rte-opi.png" width="650px">]
```
Comment (miczyg1) [inline presentations/rte-workshop.md:74]: @pietrushnic @artur-rs didn't we mention about better contrast on this picture? The lines were not visible enough to see where they point.

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: presentations/rte-workshop.md (hunk 17 above)
Comment (artur-rs) [reply presentations/rte-workshop.md:74]: @miczyg1 I have changed the image for the one with transparent background

**Review by miczyg1 — APPROVED** [review]

**Review by pietrushnic — APPROVED** [review]

**Pull request merged**